privatelink/azure/terraform/privatelink.tf

```diff
@@ -5,7 +5,7 @@ terraform {
 provider "azurerm" {
   features {
   }
-  version = "= 2.32.0"
+  version = ">= 2.55.0"
```

Comment:
We had issues before where `>=` ended up causing issues with major bumps. I think we should use `~> 2.55.0` or `>= 2.55.0, < 3.0.0` instead. https://www.terraform.io/docs/language/expressions/version-constraints.html